### ISSUE TITLE
Investigate the need for commons-beanutils dependency

### ISSUE THREAD

**Issue by ctubbsii** [apache/accumulo#1463]
In d7a407a3d6367b96fbc513f5b94b3c0511a68bf0, commons-beanutils was added to satisfy a missing runtime dependency for commons-configuration 2.5, which is not necessarily a mandatory dependency (it's optional, depending on how one uses the commons-configuration API).

It's not clear that we actually need or want to use the features in commons-configuration 2 that require commons-beanutils. In 1ed12fa585a37caf36cfd674bd7ad6903c1d4e79, some of the spammy debug logging from that library was suppressed for the testing environment, but it raises the question of whether we even need it in the first place.

For example, I believe the following code requires it:

```java
var file = new File(path);
var propsBuilder =
  new FileBasedConfigurationBuilder<PropertiesConfiguration>(PropertiesConfiguration.class)
  .configure(new Parameters().properties().setFile(file));
```

However, I think the following code would probably not require it:

```java
var file = new File(path);
var propConfig = new PropertiesConfiguration();
try (var reader = new FileReader(file)) {
    propConfig.getLayout().load(propConfig, reader);
}
```

I do not know if this code can be simply swapped out or not, and whether all our tests would pass or not, and whether we'd lose any functionality (especially important functionality, such as interpolation, importing common config from another file, or injecting system environment variables).

Investigation is needed to determine:

1. Can we safely drop the commons-beanutils dependency (with or without changing how we load properties with commons-configuration2.5's API)?
2. If yes to item 1, does interpolation work? Does injecting system properties?
3. Do we use any data type conversions that are broken?
4. Does the changes to remove it fix the spammy debug log messages that were suppressed in 1ed12fa585a37caf36cfd674bd7ad6903c1d4e79 ?

More information on commons-configuration basic features can be found at https://commons.apache.org/proper/commons-configuration/userguide/howto_basicfeatures.html ; this covers information about interpolation and data type conversions that may be relevant to understand in order to properly investigate this.

If we can drop the commons-beanutils dependency, I think we should. And, if we can get rid of the use of the `FileBasedConfigurationBuilder` class without loss of essential functionality, I think we should do that, too.

**Comment by ctubbsii** [comment]
Searching existing code for PropertiesConfiguration and FileBasedConfigurationBuilder will reveal the relevant code to change to investigate this.

**Comment by milleruntime** [comment]
I think this was inadvertently closed instead of #1465

**Comment by dlmarion** [comment]
I was able to remove the beanutils references and compile the code. Some tests are failing and I'm a little confused as to why. The tests in `start` run just fine, but some of the tests in other modules fail with:

```
ERROR] Tests run: 13, Failures: 0, Errors: 3, Skipped: 0, Time elapsed: 3.457 s <<< FAILURE! - in org.apache.accumulo.core.iterators.user.FilterTest
[ERROR] test2a(org.apache.accumulo.core.iterators.user.FilterTest)  Time elapsed: 0.081 s  <<< ERROR!
java.lang.Error: 
Unresolved compilation problems: 
	The type org.apache.commons.vfs2.FileSystemException cannot be resolved. It is indirectly referenced from required .class files
	The method getClassLoader(String) from the type ContextManager refers to the missing type FileSystemException

	at org.apache.accumulo.core.iterators.user.FilterTest.test2a(FilterTest.java:233)
```

Also, something to note is the output of `mvn dependency:tree`.  When run from `core` I see:
```
[INFO] +- org.apache.accumulo:accumulo-start:jar:2.1.0-SNAPSHOT:compile
[INFO] |  +- org.apache.commons:commons-vfs2:jar:2.6.0:compile
[INFO] |  \- commons-beanutils:commons-beanutils:jar:1.9.4:runtime
[INFO] |     \- commons-collections:commons-collections:jar:3.2.2:runtime
```

But `beanutils` has been removed as a dependency in the branch. Branch at https://github.com/dlmarion/accumulo/tree/1463-remove-beanutils

**Comment by Manno15** [comment]
I'll help investigate.

**Comment by dlmarion** [comment]
So, I ran `mvn clean install -DskipTests -DskipITs -Dspotbugs.skip` to install the snapshot jars without the dependency into my local Maven repo. `mvn clean test -Dspotbugs.skip -DskipITs` just completed successfully for me. Interesting...

**Comment by Manno15** [comment]
Yeah, I am running the full test suite currently. I got past the issue you had and so far only one IT has failed. `IteratorEnvIT`. Not sure why as of yet.

**Comment by ctubbsii** [comment]
@dlmarion So, it seems like you might be using Maven a bit wrong, and it may be the reason you're seeing inconsistent results. `mvn test` generally only works in a multi-module project if all the modules are of `jar` or `pom` packaging type. That's not the case for all Accumulo modules. You should use `mvn package` which includes the `test` phase, in order to ensure all the modules have access to their sibling artifacts. It also works if you first do `mvn install`, but I would avoid that, because that taints your local maven repository with SNAPSHOT artifacts that can cause your subsequent builds to resolve local artifacts that are stale from a previous build rather than the current sibling artifacts in your multi-module project. There are cases where you need to use `mvn install` (to make a SNAPSHOT version of a dependency available to a separate project's build), but that's an edge case. Most of your builds should use `mvn package` or `mvn verify` as your build target in the [Maven build lifecycle](https://maven.apache.org/guides/introduction/introduction-to-the-lifecycle.html).

**Comment by dlmarion** [comment]
Interesting, I was just following the [test](https://github.com/apache/accumulo/blob/main/TESTING.md) instructions. Even my IDE was unhappy and the tests failed in the same manner until I installed the artifact.

**Comment by ctubbsii** [comment]
> Interesting, I was just following the [test](https://github.com/apache/accumulo/blob/main/TESTING.md) instructions. Even my IDE was unhappy and the tests failed in the same manner until I installed the artifact.

Ugh, those instructions need to be fixed.

**Comment by ctubbsii** [comment]
This was closed by #1786
